pkg/client/client_test.go

```diff
@@ -179,148 +178,155 @@ var _ = Describe("ConmonClient", func() {
-			createRuntimeConfigWithProcessArgs(terminal, []string{"/busybox", "sleep", "10"})
-
-			sut = configGivenEnv(tmpDir, rr.runtimeRoot)
-			pid, err := sut.CreateContainer(context.Background(), &client.CreateContainerConfig{
-				ID:         ctrID,
-				BundlePath: tmpDir,
-				Terminal:   terminal,
-			})
-			Expect(err).To(BeNil())
-			Expect(pid).NotTo(Equal(0))
-			Eventually(func() error {
-				return rr.RunCommandCheckOutput(ctrID, "list")
-			}, time.Second*5).Should(BeNil())
-
-			// Start the container
-			Expect(rr.RunCommand("start", ctrID)).To(BeNil())
-
-			// Wait for container to be running
-			Eventually(func() error {
-				return rr.RunCommandCheckOutput("running", "list")
-			}, time.Second*10).Should(BeNil())
-
-			result, err := sut.ExecSyncContainer(context.Background(), &client.ExecSyncConfig{
-				ID:      ctrID,
-				Command: []string{"/busybox", "echo", "-n", "hello", "world"},
-				Timeout: 10,
-			})
+				sut = configGivenEnv(tmpDir, rr.runtimeRoot, terminal)
+				pid, err := sut.CreateContainer(context.Background(), &client.CreateContainerConfig{
+					ID:         ctrID,
+					BundlePath: tmpDir,
+					Terminal:   terminal,
+				})
+				Expect(err).To(BeNil())
+				Expect(pid).NotTo(Equal(0))
+				Eventually(func() error {
+					return rr.RunCommandCheckOutput(ctrID, "list")
+				}, time.Second*5).Should(BeNil())
 
-			Expect(err).To(BeNil())
-			Expect(result.ExitCode).To(BeEquivalentTo(0))
-			Expect(result.Stdout).To(BeEquivalentTo("hello world"))
-			Expect(result.Stderr).To(BeEmpty())
-		})
-
-		It("should set the correct exit code", func() {
-			terminal := false
-			createRuntimeConfigWithProcessArgs(terminal, []string{"/busybox", "sleep", "10"})
-
-			sut = configGivenEnv(tmpDir, rr.runtimeRoot)
-			pid, err := sut.CreateContainer(context.Background(), &client.CreateContainerConfig{
-				ID:         ctrID,
-				BundlePath: tmpDir,
-				Terminal:   terminal,
-			})
-			Expect(err).To(BeNil())
-			Expect(pid).NotTo(Equal(0))
-			Eventually(func() error {
-				return rr.RunCommandCheckOutput(ctrID, "list")
-			}, time.Second*5).Should(BeNil())
-
-			// Start the container
-			Expect(rr.RunCommand("start", ctrID)).To(BeNil())
-
-			// Wait for container to be running
-			Eventually(func() error {
-				return rr.RunCommandCheckOutput("running", "list")
-			}, time.Second*10).Should(BeNil())
-
-			result, err := sut.ExecSyncContainer(context.Background(), &client.ExecSyncConfig{
-				ID:      ctrID,
-				Command: []string{"/busybox", "invalid"},
-				Timeout: 0,
+				// Start the container
+				Expect(rr.RunCommand("start", ctrID)).To(BeNil())
+
+				// Wait for container to be running
+				Eventually(func() error {
+					return rr.RunCommandCheckOutput("running", "list")
+				}, time.Second*10).Should(BeNil())
+
+				result, err := sut.ExecSyncContainer(context.Background(), &client.ExecSyncConfig{
+					ID:       ctrID,
+					Command:  []string{"/busybox", "echo", "-n", "hello", "world"},
+					Timeout:  0,
```

Comment:
A timeout constant would be nice. I get confused whether 0 is unlimited, or it's -1.

Reply:
Added a const for that, while -1 would not be possible because of the unsigned integer.